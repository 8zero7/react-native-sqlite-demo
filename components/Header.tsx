import React from "react";
import { Text, StyleSheet } from "react-native";

interface Props {
  title: string;
}

export const Header = (props: Props) => {
  const { title } = props;
  return <Text style={styles.header}>{title}</Text>;
};

const styles = StyleSheet.create({
  header: {
    fontSize: 20,
    marginBottom: 20,
    marginTop: 20
  }
});
