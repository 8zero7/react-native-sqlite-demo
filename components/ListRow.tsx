import React from "react";
import { Text } from "react-native";
import { List } from "../types/List";

interface Props {
  list: List;
}

export const ListRow = (props: Props) => {
  return <Text>{props.list.title}</Text>;
};
