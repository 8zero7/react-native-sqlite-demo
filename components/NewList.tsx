import React from "react";

import {
  View,
  TextInput,
  StyleSheet,
  TouchableOpacity,
  Text
} from "react-native";

interface Props {
  newListTitle: string;
  handleTitleChange(title: string): void;
}

export const NewList = (props: Props) => {
  const { handleTitleChange, newListTitle } = props;
  return (
    <View style={styles.wrapper}>
      <TextInput
        placeholder="Enter a new list name"
        onChangeText={handleTitleChange}
        value={newListTitle}
        style={styles.textInput}
      />
      <TouchableOpacity style={styles.button}>
        <Text>Add List</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  textInput: {
    borderWidth: 1,
    flex: 4
  },
  wrapper: {
    flexDirection: "row"
  },
  button: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "center"
  }
});
