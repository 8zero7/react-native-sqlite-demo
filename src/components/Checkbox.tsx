import React from "react";
import { StyleSheet, Text } from "react-native";

interface Props {
  checked: boolean;
}

export const Checkbox = (props: Props) => {
  const { checked } = props;
  return <Text style={styles.check}>{checked ? "☑" : "⬜"}</Text>;
};

const styles = StyleSheet.create({
  check: {
    fontSize: 30
  }
});
