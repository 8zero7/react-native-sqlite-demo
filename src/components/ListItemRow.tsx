import React from "react";
import { Text, StyleSheet, View } from "react-native";
import { ListItem } from "../types/ListItem";

interface Props {
  listItem: ListItem;
}

export const ListItemRow = (props: Props) => {
  const { listItem } = props;
  return (
    <View style={styles.row}>
      <Text>{listItem.text}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  row: {
    borderWidth: 1,
    padding: 15,
    marginTop: 10,
    backgroundColor: "#EEE",
    borderRadius: 3,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25,
    shadowRadius: 3
  }
});
