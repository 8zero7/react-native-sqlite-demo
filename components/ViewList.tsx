import React from "react";
import {
  View,
  StyleSheet,
  Modal,
  Text,
  TouchableHighlight
} from "react-native";
import { Header } from "./Header";
import { ListItems } from "./ListItems";
import { List } from "../types/List";

interface Props {
  visible: boolean;
  list: List;
}

export const ViewList = (props: Props) => {
  const { visible, list } = props;
  if (list == null) {
    return null;
  }
  return (
    <Modal
      animationType="slide"
      transparent={false}
      visible={visible}
      onRequestClose={() => {
        // TODO: call parent
      }}
    >
      <View style={styles.container}>
        <View>
          <Header title={list.title} />

          <TouchableHighlight
            onPress={() => {
              // TODO: call parent
            }}
          >
            <Text>Hide Modal</Text>
          </TouchableHighlight>
        </View>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  container: {}
});
