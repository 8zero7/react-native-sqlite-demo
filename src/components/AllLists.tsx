/**
 * React Native SQLite Demo
 * Copyright (c) 2018-2020 Bruce Lefebvre <bruce@brucelefebvre.com>
 * https://github.com/blefebvre/react-native-sqlite-demo/blob/master/LICENSE
 */
import React, { useState } from "react";
import { View, StyleSheet, FlatList, Text, TouchableOpacity } from "react-native";

import { NewItem } from "./NewItem";
import { Header } from "./Header";
import { List } from "../types/List";
import { ListRow } from "./ListRow";
import { ViewListModal } from "./ViewListModal";
import { SettingsModal } from "./SettingsModal";
import { useLists } from "../hooks/useLists";

// Main page of the app. This component renders:
// - a header, including a cog icon to open the Settings modal
// - the form to add a new List
// - and a list of all the Lists saved locally in the app's database
export const AllLists: React.FunctionComponent = function() {
  const [newListTitle, setNewListTitle] = useState("");
  const [isListModalVisible, setIsListModalVisible] = useState(false);
  const [isSettingsModalVisible, setIsSettingsModalVisible] = useState(false);

  // Use the useLists hook to simplify list management.
  const { lists, selectList, selectedList, createList, deleteList, selectedListsItems } = useLists();

  async function handleListClicked(list: List) {
    console.log(`List clicked! Title: ${list.title}`);
    await selectList(list);
    // Open the modal which shows a single list's items
    setIsListModalVisible(true);
  }

  return (
    <View style={styles.container} testID="allListsView">
      <View style={styles.headerWithSettings}>
        <TouchableOpacity style={styles.settingsButton} onPress={() => setIsSettingsModalVisible(true)}>
          <Text style={styles.settingsButtonText}>⚙️</Text>
        </TouchableOpacity>
        <Header title="SQLite List App - with Hooks" />
      </View>

      <NewItem
        newItemName={newListTitle}
        handleNameChange={(value) => setNewListTitle(value)}
        handleCreateNewItem={createList}
        placeholderText="Enter a name for your new list"
        createButtonText="Add list"
        buttonTestId="addListButton"
        textInputTestId="newListTextInput"
      />

      <FlatList
        data={lists}
        renderItem={({ item }) => <ListRow list={item} handleListClicked={handleListClicked} />}
        keyExtractor={(item, index) => `${index}`}
      />

      <ViewListModal
        visible={isListModalVisible}
        list={selectedList}
        back={() => setIsListModalVisible(false)}
        listItems={selectedListsItems}
        refreshListItems={async function() {
          /* TODO!! */
        }}
        deleteList={deleteList}
      />

      <SettingsModal visible={isSettingsModalVisible} back={() => setIsSettingsModalVisible(false)} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingLeft: 10,
    paddingRight: 10,
    flex: 1,
  },
  headerWithSettings: {
    flexDirection: "row",
    justifyContent: "flex-start",
  },
  settingsButton: {
    marginTop: 10,
    paddingRight: 5,
    paddingBottom: 10,
    paddingTop: 10,
  },
  settingsButtonText: {
    fontSize: 20,
  },
});
