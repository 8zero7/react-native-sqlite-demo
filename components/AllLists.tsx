import React, { Component } from "react";
import { View, StyleSheet, FlatList } from "react-native";

import { NewList } from "./NewList";
import { Header } from "./Header";
import { List } from "../types/List";
import { ListRow } from "./ListRow";
import { database } from "../database/Database";
import { ViewList } from "./ViewList";

interface State {
  newListTitle: string;
  lists: List[];
  listModalVisible: boolean;
}

export class AllLists extends Component<any, State> {
  constructor(props: any) {
    super(props);
    this.state = {
      newListTitle: "",
      lists: [],
      listModalVisible: false
    };
    this.handleNewListTitleChange = this.handleNewListTitleChange.bind(this);
    this.handleCreateList = this.handleCreateList.bind(this);
    this.handleListClicked = this.handleListClicked.bind(this);
  }

  public componentDidMount() {
    this.refreshListOfLists();
  }

  public render() {
    return (
      <View style={styles.container}>
        <Header title="Bruce's List App" />
        <NewList
          newListTitle={this.state.newListTitle}
          handleTitleChange={this.handleNewListTitleChange}
          handleCreateList={this.handleCreateList}
        />
        <FlatList
          data={this.state.lists}
          renderItem={({ item }) => (
            <ListRow list={item} handleListClicked={this.handleListClicked} />
          )}
          keyExtractor={(item, index) => `${index}`}
        />
        <ViewList
          visible={this.state.listModalVisible}
          list={this.state.selectedList}
        />
      </View>
    );
  }

  private handleNewListTitleChange(title: string) {
    this.setState({
      newListTitle: title
    });
  }

  private handleCreateList(): Promise<void> {
    const { newListTitle } = this.state;
    return database.createList(newListTitle).then(() => {
      // Refresh the list of lists
      this.refreshListOfLists();
    });
  }

  private handleListClicked(list: List) {}

  private refreshListOfLists() {
    return database.getAllLists().then(lists => this.setState({ lists }));
  }
}

const styles = StyleSheet.create({
  container: {
    paddingLeft: 10,
    paddingRight: 10,
    flex: 1
  }
});
