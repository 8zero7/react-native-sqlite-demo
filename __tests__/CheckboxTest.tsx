import React, { Component } from "react";
import renderer from "react-test-renderer";

import { Checkbox } from "../src/components/Checkbox";

test("Renders correctly unchecked", () => {
  const tree = renderer.create(<Checkbox checked={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Renders correctly checked", () => {
  const tree = renderer.create(<Checkbox checked={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
