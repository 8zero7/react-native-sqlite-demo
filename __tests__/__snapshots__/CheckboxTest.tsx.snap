// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly checked 1`] = `
<Text
  style={
    Object {
      "fontSize": 30,
    }
  }
>
  ☑
</Text>
`;

exports[`Renders correctly unchecked 1`] = `
<Text
  style={
    Object {
      "fontSize": 30,
    }
  }
>
  ⬜
</Text>
`;
